### PR TITLE
Allow ERB parser to_ruby_ast to return nil

### PR DESCRIPTION
**Pull request by alexevanczuk**
## What are you trying to accomplish?
After we added types, I found this method returned `nil` on our monolith sometimes, causing `bin/packwerk update-todo` unable to execute properly.

## What approach did you choose and why?
I updated the sig to ensure it continues to work as expected. We might want to come back later and...
1) figure out when/how its `nil`
2) Add a test for when its `nil` (or possible fix it so its never `nil` if that turns out to be a better approach)

## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Non-breaking change (a change that doesn't alter functionality - i.e., code refactor, configs, etc.)

### Additional Release Notes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Include any notes here to include in the release description. For example, if you selected "breaking change" above, leave notes on how users can transition to this version.

If no additional notes are necessary, delete this section or leave it unchanged.

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] It is safe to rollback this change.

### CODE REVIEW

File: lib/packwerk/parsers/erb.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ def parse_buffer(buffer, file_path:)
         params(
           erb_ast: T.all(::AST::Node, Object),
           file_path: String
-        ).returns(::AST::Node)
+        ).returns(T.nilable(::AST::Node))
```
Comment (gmcgibbon) [inline lib/packwerk/parsers/erb.rb:47]: Is it possible to add coverage? If it isn't covered in our test suite I can see this being removed again.

Comment (alexevanczuk) [reply lib/packwerk/parsers/erb.rb:47]: Yep, I can look more into it.

Comment (alexevanczuk) [reply lib/packwerk/parsers/erb.rb:47]: It looks like ERB can contain javascript which returns `nil` rather than a node. I added a simple test to demonstrate this. I also added more types to the parser class so that the nil type is also supported by the return value of the parser classes.
